File: Data Structures & Algorithms/is-anagram/submission-0.java
class Solution {
    public boolean isAnagram(String s, String t) {
        Map<Character, Integer> map = new HashMap<>();
        for (int i=0; i<s.length(); i++) {
            if (map.containsKey(s.charAt(i))) {
                map.put(s.charAt(i), map.get(s.charAt(i))+1);
            }
            else {
                map.put(s.charAt(i),1);
            }
        }

        for (int i=0; i<t.length(); i++) {
            if (map.containsKey(t.charAt(i))) {
                map.put(t.charAt(i), map.get(t.charAt(i))-1);
            }
            else {
                return false;
            }
        }

        for (int val : map.values()) {
            if (val!=0) {
                return false;
            }
        }

        return true;
    }
}

File: Data Structures & Algorithms/reverse-string/submission-0.cpp
class Solution {
public:
    void reverseString(vector<char>& s) {
        int l = 0, r = s.size() - 1;
        char c;
        while (l < r) {
            if (s[l] != s[r]) {
                c = s[l];
                s[l] = s[r];
                s[r] = c;
            }
            l++;
            r--;
        }
    }
};